rllib/agents/a3c/a3c_tf_policy.py

```diff
@@ -1,178 +1,181 @@
+                max_seq_len = tf.reduce_max(train_batch[SampleBatch.SEQ_LENS])
+                valid_mask = tf.sequence_mask(train_batch[SampleBatch.SEQ_LENS], max_seq_len)
+                valid_mask = tf.reshape(valid_mask, [-1])
+            else:
+                valid_mask = tf.ones_like(train_batch[SampleBatch.REWARDS])
+
+            log_prob = action_dist.logp(train_batch[SampleBatch.ACTIONS])
+            vf = model.value_function()
+
+            # The "policy gradients" loss
+            self.pi_loss = -tf.reduce_sum(
+                tf.boolean_mask(log_prob * train_batch[Postprocessing.ADVANTAGES], valid_mask)
+            )
+
+            delta = tf.boolean_mask(vf - train_batch[Postprocessing.VALUE_TARGETS], valid_mask)
+
+            # Compute a value function loss.
+            if self.config.get("use_critic", True):
+                self.vf_loss = 0.5 * tf.reduce_sum(tf.math.square(delta))
+            # Ignore the value function.
+            else:
+                self.vf_loss = tf.constant(0.0)
+
+            self.entropy_loss = tf.reduce_sum(
+                tf.boolean_mask(action_dist.entropy(), valid_mask))
+
+            self.total_loss = (
+                self.pi_loss + self.vf_loss * self.config["vf_loss_coeff"] - self.entropy_loss * self.entropy_coeff
+            )
+
+            return self.total_loss
+
+        @override(base)
+        def stats_fn(self, train_batch: SampleBatch) -> Dict[str, TensorType]:
+            return {
+                "cur_lr": tf.cast(self.cur_lr, tf.float64),
+                "entropy_coeff": tf.cast(self.entropy_coeff, tf.float64),
+                "policy_loss": self.pi_loss,
+                "policy_entropy": self.entropy_loss,
+                "var_gnorm": tf.linalg.global_norm(
+                    list(self.model.trainable_variables())),
+                "vf_loss": self.vf_loss,
+            }
+
+        @override(base)
+        def grad_stats_fn(
+            self, train_batch: SampleBatch, grads: ModelGradients
+        ) -> Dict[str, TensorType]:
+            return {
+                "grad_gnorm": tf.linalg.global_norm(grads),
+                "vf_explained_var": explained_variance(
+                    train_batch[Postprocessing.VALUE_TARGETS],
+                    self.model.value_function()
+                ),
+            }
+
+        @override(base)
+        def postprocess_trajectory(
+            self, sample_batch, other_agent_batches=None, episode=None
+        ):
+            sample_batch = super().postprocess_trajectory(sample_batch)
+            return compute_gae_for_sample_batch(
+                self, sample_batch, other_agent_batches, episode
+            )
+
+        @override(base)
+        def compute_gradients_fn(
+            self, optimizer: LocalOptimizer, loss: TensorType
+        ) -> ModelGradients:
+            return compute_gradients(self, optimizer, loss)
+
+    return A3CTFPolicy
+
+
+A3CDynamicTFPolicy = get_a3c_tf_policy(DynamicTFPolicyV2)
```

Comment:
let's rename this A3CStaticGraphTFPolicy now.

Reply:
Ah, yeah, sorry. Didn't care about my own suggestion :D